monkey/infection_monkey/exploit/drupal.py

```diff
@@ -0,0 +1,175 @@
+"""
+Remote Code Execution on Drupal server - CVE-2019-6340
+Implementation is based on:
+    https://gist.github.com/leonjza/d0ab053be9b06fa020b66f00358e3d88/f9f6a5bb6605745e292bee3a4079f261d891738a.
+"""
+
+import logging
+import re
+import requests
+from urllib.parse import urljoin, urlparse
+from infection_monkey.exploit.web_rce import WebRCE
+
+__author__ = 'Ophir Harpaz'
+
+LOG = logging.getLogger(__name__)
+
+
+def remove_port(url):
+    parsed = urlparse(url)
+    with_port = f'{parsed.scheme}://{parsed.netloc}'
+    without_port = re.sub(':[0-9]+$', '', with_port)
+    return without_port
+
+
+def build_url(*args) -> str:
+    f = ''
+    for x in args:
+        f = urljoin(f, x)
+    return f
+
+
+def check_drupal_cache(r: requests.Response) -> bool:
+    """
+    Check if a response had the cache header.
+    """
+    return 'X-Drupal-Cache' in r.headers and r.headers['X-Drupal-Cache'] == 'HIT'
```

Comment:
I wonder if this shouldn't be part of a drupal fingerprinter, to better focus the exploitation?

@VakarisZ WDYT?

Reply:
This is an integral part of this exploiter and the result is per-article (some might be cached, some aren't) - not sure that it belongs in the fingerprinter.